@xen-orchestra/xapi/network.mjs

```diff
@@ -6,9 +8,35 @@ export default class Network {
     const pluggedVIFs = network.$VIFs.filter(vif => vif.currently_attached && vif.$VM.power_state === 'Running')
     const pluggedPIFs = network.$PIFs.filter(pif => pif.currently_attached)
 
-    await Promise.allSettled(pluggedVIFs.map(vif => this.callAsync('VIF.unplug', vif.$ref)))
-    await Promise.allSettled(pluggedPIFs.map(pif => this.callAsync('PIF.plug', pif.$ref)))
-    await Promise.allSettled(pluggedPIFs.map(pif => this.callAsync('PIF.plug', pif.$ref)))
-    await Promise.allSettled(pluggedVIFs.map(vif => this.callAsync('VIF.plug', vif.$ref)))
+    let errorOccurred = false
+
+    for (const { method, networkInterfaces } of [
```

Comment:
I would put directly the ref in networkInterfaces, with something like `.map(({$ref})=>$ref)`